docs/specs/servers/notify/notify-server-api.md

```diff
@@ -2,183 +2,244 @@
 
 ## Authentication
 
-All endpoints expect an `Authorization` header in the form `Authorization: Bearer <project_secret>` using the project secret associated with a project ID. The secret used should be the one that was generated automatically when configuring notify - with the name `notify_api_secret`.
+All endpoints expect an `Authorization` header in the form `Authorization: Bearer <notify_api_secret>` using the notify_api_secret associated with a project ID. The secret used should be the one that was generated automatically when configuring notify - with the name `notify_api_secret`.
 
-## Notify
+## Send Notifications
 
-Send a notification to a set of accounts.
+Send 1 or more notifications to accounts. Notifications handled itempotently are deduplicated by `notificationId`.
 
-`POST /notify`
-
-Body:
+`POST /v1/<project-id>/notify`
 
 ```typescript
-{
-  "notification": {
-    "title": string,
-    "body": string,
-    "icon": string,
-    "url": string,
-    "type": string,
+type Body = [{
+  // Idempotency key and ID to track status
+  notificationId?: string | null,
```

Comment:
Do we really need to give the ability to provide the ID for a new notification? For me, server-generated IDs look more consistent, and does the client need this ability? cc @chris13524

Reply:
The primary purpose of this ID is actually for idempotency.

We can have a idempotencyKey field instead, and require the client to read the response in order to call the status endpoints. But I figured it's a slightly nicer API for the caller to be able to specify their own IDs in order to avoid storing the response to their database if they already have notification IDs on their end.

In my approach this notificationId would be dapp-facing only, unique key being separate per-dapp, and the pk of the database would still be a server-generated UUID. Happy with either approach really, but we need idempotency.